src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
@WilliamBZA I feel we're talking past each other. I fully agree on hiding the source of the endpoint information, this doesn't change the fact that I propose to not push endpoint by endpoint, but a merged (concat) single value (an array of endpoints) instead. Or alternatively, to decouple the updated values from changing the scope values directly to prevent excessive re-rendering.